hadoop-hdds/docs/content/interface/Cli.md

```diff
@@ -0,0 +1,209 @@
+The object can be a volume, bucket or a key. The action is various verbs like
+create, list, delete etc.
+
+Ozone URL can point to a volume, bucket or keys in the following format:
+
+_\[schema\]\[server:port\]/volume/bucket/key_
+
+
+Where,
+
+1. **Schema** - This should be `o3` which is the native RPC protocol to access
+  Ozone API. The usage of the schema is optional.
+
+2. **Server:Port** - This is the address of the Ozone Manager. If the port is
+omitted the default port from ozone-site.xml will be used.
+
+Depending on the call, the volume/bucket/key names will be part of the URL.
+Please see volume commands, bucket commands, and key commands section for more
+detail.
+
+## Volume operation
+
+Volume is the top level element of the hierarchy and supposed to managed only by the administrators. Quota and the owner user can be defined.
+
+Example commands:
+
+```shell
+> ozone sh volume create /vol1
+```
+
+```shell
+bash-4.2$ ozone sh volume info /vol1
+{
+  "metadata" : { },
+  "name" : "vol1",
+  "admin" : "hadoop",
+  "owner" : "hadoop",
+  "creationTime" : "2020-07-28T12:31:50.112Z",
+  "modificationTime" : "2020-07-28T12:31:50.112Z",
+  "acls" : [ {
+    "type" : "USER",
+    "name" : "hadoop",
+    "aclScope" : "ACCESS",
+    "aclList" : [ "ALL" ]
+  }, {
+    "type" : "GROUP",
+    "name" : "users",
+    "aclScope" : "ACCESS",
+    "aclList" : [ "ALL" ]
+  } ],
+  "quota" : 1152921504606846976
+}
+```
+
+```shell
+ozone sh volume list /
+{
+  "metadata" : { },
+  "name" : "s3v",
+  "admin" : "hadoop",
+  "owner" : "hadoop",
+  "creationTime" : "2020-07-27T11:32:22.314Z",
+  "modificationTime" : "2020-07-27T11:32:22.314Z",
+  "acls" : [ {
+    "type" : "USER",
+    "name" : "hadoop",
+    "aclScope" : "ACCESS",
+    "aclList" : [ "ALL" ]
+  }, {
+    "type" : "GROUP",
+    "name" : "users",
+    "aclScope" : "ACCESS",
+    "aclList" : [ "ALL" ]
+  } ],
+  "quota" : 1152921504606846976
+}
+....
+```
+## Bucket operation
+
+Bucket is the second level hierarchy, similar to the AWS S3 buckets. Users can create buckets in volumes with permissions.
+
+Command examples:
+
+```shell
+> ozone sh bucket create /vol1/bucket1
+```shell
+
+```shell
+> ozone sh bucket info /vol1/bucket1
+{
+  "metadata" : { },
+  "volumeName" : "vol1",
+  "name" : "bucket1",
+  "storageType" : "DISK",
+  "versioning" : false,
+  "creationTime" : "2020-07-28T13:14:45.091Z",
+  "modificationTime" : "2020-07-28T13:14:45.091Z",
+  "encryptionKeyName" : null,
+  "sourceVolume" : null,
+  "sourceBucket" : null
+}
+```
+
+Bucket is the level of the [Transparent Data Encryption]({{< ref "security/SecuringTDE.md" >}})
+
+## Key
+
+Key is the object which can store the data.
+
+```shell
+>ozone sh key put /vol1/bucket1/README.md README.md
+```
+
+<div class="alert alert-warning" role="alert">
+
+The standard `ozone sh _object_type _action_ _url_` scheme can cause some confusion here as the parameter order is `ozone sh key put <DESTINATION> <SOURCE>` instead of the opposite. 
```

Comment:
```suggestion
In this case the standard `ozone sh <object_type> <action> <url>` scheme may be a bit confusing at first, as it results in the syntax `ozone sh key put <destination> <source>` instead of the arguably more natural order of `<source> <destination>`.
```